File: src/main/kotlin/space/davids_digital/kiri/rest/dto/telegram/TelegramDonationDto.kt
package space.davids_digital.kiri.rest.dto.telegram

import java.time.OffsetDateTime

data class TelegramDonationDto (
    val telegramPaymentChargeId: String,
    val providerPaymentChargeId: String,
    val chatId: Long,
    val chatTitle: String?,
    val chatUsername: String?,
    val userId: Long?,
    val userUsername: String?,
    val userFirstName: String?,
    val currency: String,
    val totalAmount: Int,
    val invoicePayload: String,
    val paidAt: OffsetDateTime,
    val refunded: Boolean,
    val refundedAt: OffsetDateTime?,
    val messageId: Int,
)

data class TelegramDonationsSummaryDto (
    val totalReceived: Long,
    val totalRefunded: Long,
    val net: Long,
    val countTotal: Long,
    val countRefunded: Long,
    val currency: String,
)

File: admin-ui/lib/api/types/telegram/TelegramDonationDto.ts
export type TelegramDonationDto = {
    telegramPaymentChargeId: string;
    providerPaymentChargeId: string;
    chatId: number;
    chatTitle: string | null;
    chatUsername: string | null;
    userId: number | null;
    userUsername: string | null;
    userFirstName: string | null;
    currency: string;
    totalAmount: number;
    invoicePayload: string;
    paidAt: number;
    refunded: boolean;
    refundedAt: number | null;
    messageId: number;
};

export type TelegramDonationsSummaryDto = {
    totalReceived: number;
    totalRefunded: number;
    net: number;
    countTotal: number;
    countRefunded: number;
    currency: string;
};
